anthropic-java-bedrock/src/main/kotlin/com/anthropic/bedrock/credentials/BedrockCredentials.kt

```diff
@@ -0,0 +1,391 @@
+                    return serialize().size.toLong()
+                }
+
+                override fun repeatable(): Boolean = true
+
+                override fun close() {}
+            }
+        }
+
+        /**
+         * Creates a JSON [ObjectNode] representing the JSON data parsed from a
+         * [HttpRequestBody].
+         *
+         * @param body A new request body holding the JSON data.
+         *
+         * @return The JSON object data to form the request body, or `null` if
+         *     the request does not have a body or if the body is empty.
+         */
+        // NOTE: "internal" visibility allows this method to be used in unit
+        // tests to create test fixtures.
+        @JvmSynthetic
+        internal fun bodyToJson(body: HttpRequestBody?): ObjectNode? {
+            val jsonData = ByteArrayOutputStream()
+
+            body?.writeTo(jsonData)
+            if (jsonData.size() > 0) {
+                return ObjectMapper().readValue(
+                    jsonData.toByteArray(), ObjectNode::class.java)
+            }
+            return null
+        }
+    }
+
+    /**
+     * Gets the base URL for the Amazon Bedrock runtime service. The base URL
+     * includes the Bedrock runtime service name and the AWS region.
+     *
+     * @return The base URL for the Bedrock runtime service.
+     */
+    override fun baseUrl(): String =
+        // Could use the AWS "BedrockEndpointProvider" in "fromEnv()", but that
+        // is a large dependency that brings in all the competing AWS SDK
+        // classes for Bedrock that are not desired in this SDK.
+        "https://$SERVICE_NAME.$region.amazonaws.com"
+
+    /**
+     * Prepares the request for use with Amazon Bedrock.
+     *
+     * A number of changes are made to support the requirements of AWS requests
+     * to the Bedrock service:
+     *
+     * * The model ID is moved from the JSON body content to the path segments.
+     * * The Anthropic version is added to the JSON body content.
+     * * The AWS action to be performed is added to the path segments.
+     *
+     * @param request The request to prepare. This request will not be modified.
+     *
+     * @return A new request, copied from the given request, and then prepared
+     *     and modified to conform to the Bedrock requirements.
+     *
+     * @throws AnthropicException If the path segments describe an operation
+     *     that is not yet supported by Anthropic models hosted on the Bedrock
+     *     service. If the JSON body is not present. If the request has already
+     *     been prepared.
+     */
+    override fun prepare(request: HttpRequest): HttpRequest {
+        val json: ObjectNode? = bodyToJson(request.body)
+        val modelId: String
+        val isStreaming: Boolean
+
+        // Interpret and modify the JSON body.
+        if (json != null) {
+            val model = json.remove("model")
+                ?: throw AnthropicException("No model found in body.")
```

Comment:
Here and throughout, for any "invalid data" related issues, we probably want to use `AnthropicInvalidDataException`